### PR TITLE
Max index length

### LINKED ISSUE #217: BR cannot restore when max-index-length>3072
Please answer these questions before submitting your issue. Thanks!

1. What did you do?
   1. set tidb config `max-index-length` to 12288;
   2. `create table t1 (a varchar(3072) primary key);`
   3. br backup full -s local:///home/tidb/tmp/br
   4. drop table t1;
   5. br restore full -s local:///home/tidb/tmp/br 

2. What did you expect to see?
restore success

3. What did you see instead?
```Error: [ddl:1071]Specified key was too long; max key length is 3072 bytes```

4. What version of BR and TiDB/TiKV/PD are you using?
br master
tidb master

### PR DESCRIPTION
**Pull request by 3pointer**
### What problem does this PR solve? 
Fix #217 

### What is changed and how it works?
add flag to set max-index-length when restoring data

### Check List 

Tests 

 - Manual test (according to #217)

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

File: pkg/task/restore.go [hunk 1]
```diff
@@ -46,18 +48,21 @@ const (
...
+	NoSchema       bool `json:"no-schema" toml:"no-schema"`
+	MaxIndexLength int  `json:"max-index-length" toml:"max-index-length"`
 }
 
 // DefineRestoreFlags defines common flags for the restore command.
 func DefineRestoreFlags(flags *pflag.FlagSet) {
 	// TODO remove experimental tag if it's stable
 	flags.Bool(flagOnline, false, "(experimental) Whether online when restore")
 	flags.Bool(flagNoSchema, false, "skip creating schemas and tables, reuse existing empty ones")
+	flags.Int(flagMaxIndexLength, 3072, "need keep this as same as tidb max-index-length")
```
Comment (kennytm) [inline pkg/task/restore.go]:
```suggestion
	flags.Int(flagMaxIndexLength, 3072, "value of max-index-length setting of the restored TiDB")
```

File: pkg/task/restore.go [hunk 2]
```diff
@@ -148,6 +149,14 @@ func RunRestore(c context.Context, g glue.Glue, cmdName string, cfg *RestoreConf
 		return err
 	}
 	// execute DDL first
+
+	// set max-index-length before execute DDLs and create tables
+	// we set this value to max(3072*4), otherwise we might not restore table
+	// when upstream and downstream both set this value greater than default(3072)
+	conf := config.GetGlobalConfig()
+	conf.MaxIndexLength = config.DefMaxOfMaxIndexLength
+	config.StoreGlobalConfig(conf)
```
Comment (5kbpers) [inline pkg/task/restore.go:158]: Better to log a warning or explain it in our document.

Comment (3pointer) [reply pkg/task/restore.go:158]: okay，add a warn log

**Pull request merged**